packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -301,16 +301,88 @@ def _create_mint_batch_tx(
     ) -> str:
         """Mint a batch of items."""
         # mint batch
+        for i in range(len(batch_mint_quantities)):
+            decoded_type = Helpers().decode(self.token_ids[i])
+            assert (
+                decoded_type == 1 or decoded_type == 2
+            ), "The prefix type must be 1 or 2."
+            if decoded_type == 1:
+                assert (
+                    batch_mint_quantities[i] == 1
+                ), "Cannot mint NFT with mint_quantity more than 1"
+        data = b"MintingBatch"
         nonce = ledger_api.api.eth.getTransactionCount(
             ledger_api.api.toChecksumAddress(deployer_address)
         )
         nonce += 1
         tx = self.instance.functions.mintBatch(
-            recipient_address, self.token_ids, batch_mint_quantities
+            recipient_address, self.token_ids, batch_mint_quantities, data
         ).buildTransaction(
             {
                 "chainId": 3,
-                "gas": 300000,
+                "gas": 500000,
+                "gasPrice": ledger_api.api.toWei("50", "gwei"),
+                "nonce": nonce,
+            }
+        )
+
+        return tx
+
+    def get_mint_single_tx(
+        self,
+        deployer_address: Address,
+        recipient_address: Address,
+        mint_quantity: int,
+        ledger_api: LedgerApi,
+        skill_callback_id: ContractId,
+    ) -> TransactionMessage:
+
+        tx = self._create_mint_single_tx(
+            deployer_address=deployer_address,
+            recipient_address=recipient_address,
+            token_id=self.game_currency_id,
```

Comment:
this needs to be updated and replaced with token_id, an argument to the function